exercise3.js

```diff
@@ -85,32 +127,116 @@
+    })
+  return Math.max(...nums);
+};
 
+max(numbers)
 // 7.b -Write a function that sorts the given numbers array.  Allow the function to sort the array in descending order
 
-function sortNums(numbers,desc=false) {
-    //your code...
+
+let sorted = (arr, mer) => {
+  let nums = [];
+  arr.map((item) => {
+      //same as above
+    if (item == 'one') {
+      nums.push(1);
+    }
+    if (item == 'two') {
+      nums.push(2);
+    }
+    if (item == '3') {
+      nums.push(3);
+    }
+    if (item == 'three') {
+      nums.push(3);
+    }
+    if (typeof item === 'number') {
+      nums.push(item);
+    }
+  });
+  console.log(mer(nums));
 };
 
+//my implementation of merge sort but it does it backwards
+let reverseMergeSort = (arr) => {
+    //checks if array length is 1 if so returns it
+  if (arr.length === 1) {
+    return arr;
+  }
+  //gets middle index
+  let middle = Math.floor(arr.length / 2);
+  //cuts left side
+  let left = arr.slice(0, middle);
+  //cuts right side
+  let right = arr.slice(middle);
+  //recursevly  does above function
+  let sortedLeft = reverseMergeSort(left);
+  let sortedRight = reverseMergeSort(right);
+  return merge(sortedLeft, sortedRight);
+};
+//helper function to main function
+let merge = (left, right) => {
+
+  let results = [];
+  let leftIndex = 0;
+  let rightIndex = 0;
+  //while the left Index is less than left side so until its fully splitted 
+  while (leftIndex < left.length && rightIndex < right.length) {
+      //if number in the left is less than number in the right push right THIS IS WHAT PRETTY MUCH MAKES IT GO FROM BIG TO LOW
+    if (left[leftIndex] < right[rightIndex]) {
+      results.push(right[rightIndex]);
+      rightIndex++;
+    } else {
+        //SECOND PART OF PUSHING 
+      results.push(left[leftIndex]);
+      leftIndex++;
+    }
+  }
+  //start adding all the pieces that were splittled above
+  return results.concat(left.slice(leftIndex)).concat(right.slice(rightIndex));
+};
 
+sorted(numbers, reverseMergeSort);
 
 
 /************************************************************* */
 // Problem 8:
 //add an example of at least 5 JavaScript data types to the given mapObj.  The key is the example data type, and the value is the name of the data type.  An object data type has already been set as the 1st key / val pair.
 
 const mapObj = new Map();
-mapObj.set({company : "TEKsystems"},"object");
-
-
-console.log(mapObj.has({company : "TEKsystems"}));  
+//adds valyes to the Map
+mapObj.set('object', { company: 'TEKsystems' });
+mapObj.set(`string`, `string`);
+mapObj.set(`number`, 5);
+mapObj.set(`boolean`, true);
+//arrays are technically objects
+mapObj.set('array', []);
+
+   
 //The above console.log() statmeent returns false.  Write another console.log() statement explaining why this line of code prints false.  Refactor the code on line 106, so you can successfully check to see if {company : "TEKsystems"} exists in the mapObj.
-
+console.log(mapObj.has('object')); 
```

Comment:
this does pass the test.  but it doesn't check to see exactly if the object 
![image](https://user-images.githubusercontent.com/30554629/84024164-5dd96a80-a957-11ea-8b3a-92a6d885f53f.png)
exists in the Map().  Any object could be in the map, and it would pass the test.  And to be honest, you could even put **any data type as the value**, and it would also pass the test.  

The solution is, you set it to a variable, then check if that variable is in the map, then that would work.